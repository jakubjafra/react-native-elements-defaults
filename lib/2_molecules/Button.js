const _ = require('lodash');
const utils = require('../utils');

// TODO: Make this function immutable.
function override(defaultStyles) {
    const __oldModule__ = require('react-native-elements/src/buttons/Button');
    const __oldModuleDefault__ = __oldModule__.default;

    const Button = function (props) {
        const newProps = {};

        newProps.style = utils.overrideBottom(props.style, defaultStyles.style);
        newProps.containerViewStyle = utils.overrideBottom(props.containerViewStyle, defaultStyles.containerViewStyle);
        newProps.buttonStyle = utils.overrideBottom(props.buttonStyle, defaultStyles.buttonStyle);
        newProps.textStyle = utils.overrideBottom(props.textStyle, defaultStyles.textStyle);
        newProps.disabledStyle = utils.overrideBottom(props.disabledStyle, defaultStyles.disabledStyle);
        newProps.activityIndicatorStyle = utils.overrideBottom(props.activityIndicatorStyle, defaultStyles.activityIndicatorStyle);

        return __oldModuleDefault__.call(undefined, _.defaultsDeep(newProps, props));
    };

    Button.propTypes = __oldModuleDefault__.propTypes;
    Button.defaultProps = __oldModuleDefault__.defaultProps;
    __oldModule__.default = Button;
}

module.exports = override;
