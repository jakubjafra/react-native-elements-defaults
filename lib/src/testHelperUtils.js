import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name,
                      __oldElement__,
                      __element__,
                      basicTheme,
                      createElement = element => React.createElement(element, null, null)) {
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
