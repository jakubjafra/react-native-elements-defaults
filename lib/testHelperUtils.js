import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name, __oldElement__, __element__, basicTheme){
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(<OldElement />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(<Element />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(<OldElement />).toJSON();
            const tree = renderer.create(<Element />).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(<OldElement />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(<Element />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(<OldElement />).toJSON();
            const tree = renderer.create(<Element />).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
