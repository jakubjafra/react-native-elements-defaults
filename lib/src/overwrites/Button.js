const utils = require('../utils');
const functionalComponentCompose = require('./_funcionalComponentCompose');

const COMPOSE_NAME = 'Button';
const compose = functionalComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/buttons/Button'),
        compose,
        true,
    ),
};
