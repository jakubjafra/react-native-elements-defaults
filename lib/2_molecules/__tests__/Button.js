import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import * as utils from '../../testHelperUtils';

describe('Button', () => {
    utils.defaultTests(
        'Button',
        require('react-native-elements/src/buttons/Button'),
        require('../Button'),
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );
});
