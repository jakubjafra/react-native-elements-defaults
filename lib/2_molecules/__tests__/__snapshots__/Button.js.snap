// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with basic theme applied Button renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Button with basic theme applied OldButton renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Button with no theme applied Button renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Button with no theme applied OldButton and Button returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`Button with no theme applied OldButton renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;
